### PR TITLE
Bugbash/T-iOS-204

### PR DESCRIPTION
**Pull request by Rimktarii**
Adding Delete anyway bottomsheet when a file is in a report

### CODE REVIEW

File: Tella/Scenes/HomeView/ViewModel/FileListViewModel.swift [hunk 1]
```diff
@@ -367,6 +367,23 @@ extension FileListViewModel {
...
     }
     
+    
+    func checkReportsBeforeDelete() -> Bool {
+        // check reports before deleting files
+        
+        let fileIds = selectedFiles.compactMap { $0.id }
+        let fileIsInReport = appModel.tellaData?.database.checkFilesInReports(ids: fileIds)
+
+        if fileIsInReport == true {
```
Comment (ValbuenaG) [inline Tella/Scenes/HomeView/ViewModel/FileListViewModel.swift:377]: what do you think about changing this conditional to this:

```
if !filesInReport {
  deleteSelectedFiles()
}

return filesInReport

```

File: Tella/Supporting Files/ar.lproj/Localizable.strings [hunk 2]
```diff
@@ -78,6 +78,9 @@
 "Vault_RenameFile_SheetTitle" = "إعادة تسمية الملف";
 "Vault_RenameFile_Cancel_SheetAction" = "CANCEL";
 "Vault_RenameFile_Save_SheetAction" = "SAVE";
+"Vault_Confirm_delete_Description" = "Some of the selected files are currently used in your connections. Deleting may impact forms and reports that haven’t been sent yet";
```
Comment (dhekra-rouatbi) [inline Tella/Supporting Files/ar.lproj/Localizable.strings:81]: What do you think if you change these lacalizable to :
`"Vault_DeleteFile_Warning_SheetExpl"` 
`"Vault_DeleteFile_Warning_SheetTitle" `
`"Vault_DeleteFile_DeleteAnyway_SheetAction"  `

**Review by dhekra-rouatbi — CHANGES_REQUESTED** [review]

File: Tella/Data/Database/TellaDataBase.swift [hunk 3]
```diff
@@ -84,6 +83,22 @@ class TellaDataBase : DataBase {
         
     }
     
+    func checkFilesInReports(ids: [String]) -> Bool {
+        var isFileInReport = false
+        do {
+            try ids.forEach({ vaultId in
+                let condition = [KeyValue(key: D.cVaultFileInstanceId, value: vaultId)]
+                let responseDict = try statementBuilder.selectQuery(tableName: D.tReportInstanceVaultFile,
```
Comment (dhekra-rouatbi) [inline Tella/Data/Database/TellaDataBase.swift:91]: I think this condition will check only the files in the reports, we need to add an other condition for Uwazi files

File: Tella/Data/Database/TellaDataBase.swift [hunk 4]
```diff
@@ -84,6 +83,22 @@ class TellaDataBase : DataBase {
...
+                                                                    andCondition: condition)
+                isFileInReport = !responseDict.isEmpty
```
Comment (dhekra-rouatbi) [inline Tella/Data/Database/TellaDataBase.swift:93]: maybe this will return only the result of the last file ?

**Review by dhekra-rouatbi — CHANGES_REQUESTED** [review]

File: Tella/Scenes/HomeView/ViewModel/FileListViewModel.swift [hunk 5]
```diff
@@ -367,6 +367,23 @@ extension FileListViewModel {
...
+            // Show pop up
+            return  true
+        } else {
+            deleteSelectedFiles()
```
Comment (dhekra-rouatbi) [inline Tella/Scenes/HomeView/ViewModel/FileListViewModel.swift:381]: I think `deleteSelectedFiles()` will directly delete the file after user clicks on delete file

**Review by dhekra-rouatbi — CHANGES_REQUESTED** [review]

File: Tella/Scenes/HomeView/ViewModel/FileListViewModel.swift [hunk 6]
```diff
@@ -367,6 +367,23 @@ extension FileListViewModel {
         }
     }
     
+    
+    func checkReportsBeforeDelete() -> Bool {
```
Comment (dhekra-rouatbi) [inline Tella/Scenes/HomeView/ViewModel/FileListViewModel.swift:371]: what do you think if we change it to this ?

    func filesAreUsedInConnections() -> Bool {
        guard let database = appModel.tellaData?.database else { return false }
        let fileIds = selectedFiles.compactMap { $0.id }
        return database.checkFilesInReports(ids: fileIds)
    }

**Review by dhekra-rouatbi — CHANGES_REQUESTED** [review]

File: Tella/Scenes/HomeView/Views/FileListView/File list/MoreFileActionButton.swift [hunk 7]
```diff
@@ -137,18 +141,34 @@ struct MoreFileActionButton: View {
...
         })
     }
     
+    private func deleteAction() {
+        fileListViewModel.deleteSelectedFiles()
+        fileListViewModel.selectingFiles = false
+        fileListViewModel.resetSelectedItems()
+    }
+    
+    func showDeleteAnywaySheet() {
```
Comment (dhekra-rouatbi) [inline Tella/Scenes/HomeView/Views/FileListView/File list/MoreFileActionButton.swift:158]: maybe `showDeleteWarningSheet` is better ? what do you think ?

**Review by dhekra-rouatbi — CHANGES_REQUESTED** [review]

File: Tella/Supporting Files/ar.lproj/Localizable.strings (hunk 2 above)
Comment (Rimktarii) [reply Tella/Supporting Files/ar.lproj/Localizable.strings:81]: and maybe this  "Vault_DeleteFile_Warning_SheetDescription" instead this "Vault_DeleteFile_Warning_SheetExpl"

**Pull request merged**